libsolidity/lsp/HandlerBase.h

```diff
@@ -37,6 +39,9 @@ class HandlerBase
 {
 public:
 	explicit HandlerBase(LanguageServer& _server): m_server{_server} {}
+	virtual ~HandlerBase() = default;
+
+	virtual void initialize(Json::Value const& /*_clientCapabilities*/, Json::Value& /*_replyCapabilities*/) {}
```

Comment:
This does not look like generic initialization. Looks like hook for modifying the capability list before it's reported to the client:

```suggestion
	virtual void onReportCapabilities(Json::Value const& /*_clientCapabilities*/, Json::Value& /*_replyCapabilities*/) {}
```

Also, is `_replyCapabilities` always empty or pre-initialized? This should be specified in the docstring.

Finally, I think it would make more sense for the function to only return the list of *extra* capabilities the handler provides instead of being able to modify the whole list. Should one handler really be able to remove capabilities added by another handler?

Reply:
done